prebid-server/pbs-modules/rules-engine.md

```diff
@@ -0,0 +1,805 @@
+---
+layout: page_v2
+sidebarType: 5
+title: Prebid Server | Modules | Rules Engine
+---
+
+# Prebid Server | Modules | Rules Engine
+{: .no_toc}
+
+- TOC
+{:toc}
+
+## Overview
+
+The Rules Engine is a flexible module that allows host companies to define JSON configurations for controlling various aspects of requests, rather than having to code exceptions in a standalone module or in the core of Prebid Server.
+
+Most of the use cases for this version are centered around controlling which bid adapters are called in particular scenarios, based on:
+
+- host geography
+- user geography
+- presence of a synced ID for that bidder
+- presence of first party data
+- channel (web vs mobile app)
+
+### Limitations
+
+The first phase is limited in several ways:
+
+- Runs only at the Processed-Auction stage.
+- Only delivers functions for including and excluding bidders.
+- Does not necessarily reload config periodically like the Floors feature, making integration with a backend machine learning system more challenging.
+- Does not support getting rules config from multiple sources. e.g. host-level rules cannot be merged with account-level rules. And ideally, there could be an ecosystem of vendors supplying config for different rule sets. This is not currently possible.
+
+### Future Enhancements
+
+- Add additional results functions.
+- Support additional module stages.
+- Improve getting the configuration from multiple sources.
+
+The team will tackle enhancements as important use cases drive priority. 
+
+## Quick Start
+
+For those who prefer to jump right in and see what it looks like, here's an example Rules Engine configuration that:
+
+98% of the time
+
+- removes bidderA in the EU datacenter
+- removes bidderD in the EU datacenter if that bidder hasn't synced with this user
+- removes bidderB and bidderF in the APAC datacenter
+- logs any bidder removed to analytics
+
+2% of the time
+
+- logs "control" to analytics
+
+```json
+{
+  "hooks": {
+    "modules": {
+      "pb-rules-engine": {
+        "enabled": true,
+        "timestamp": "20250314 00:00:00",
```

Comment:
This is not currently implemented in Go but I think we should leave it in the docs and we will support it. Right now Go is hashing the fetched config and comparing it against a cached hash to determine if there is a change but we will add timestamp support.